Sources/XMTPiOS/Group.swift

```diff
@@ -0,0 +1,106 @@
+//
+//  Group.swift
+//
+//
+//  Created by Pat Nakajima on 2/1/24.
+//
+
+import Foundation
+import LibXMTP
+
+public struct Group: Identifiable, Equatable, Hashable {
+	var ffiGroup: FfiGroup
+	var client: Client
+
+	public struct Member {
+		var ffiGroupMember: FfiGroupMember
+
+		public var accountAddress: String {
+			ffiGroupMember.accountAddress
+		}
+	}
+
+	public var id: Data {
+		ffiGroup.id()
+	}
+
+	public static func == (lhs: Group, rhs: Group) -> Bool {
+		lhs.id == rhs.id
+	}
+
+	public func hash(into hasher: inout Hasher) {
+		id.hash(into: &hasher)
+	}
+
+	public func members() async throws -> [Member] {
+		_ = try await ffiGroup.sync()
+		return try ffiGroup.listMembers().map(\.fromFFI)
+	}
+
+	public var cachedMembers: [Member] {
+		do {
+			return try ffiGroup.listMembers().map(\.fromFFI)
+		} catch {
+			return []
+		}
+	}
+
+	public func addMembers(addresses: [String]) async throws {
+		try await ffiGroup.addMembers(accountAddresses: addresses)
+		try await ffiGroup.sync()
```

Comment:
I think calling sync in all these places is probably hurting the performance? I just expose the sync method instead.